File: java/my_java_dojo/cyber-dojo/anagrams/Kata1Test.java
package tw.idv.ken.kata.anagrams;

import org.junit.After;
import org.junit.Before;
import org.junit.Test;

import java.util.Arrays;

import static org.junit.Assert.*;

public class Kata1Test {
    private Kata1 instance = new Kata1();

    @Before
    public void setUp() {
    }

    @After
    public void tearDown() {
    }

    @Test
    public void test_2_letters() {
        String[] result = instance.anagrams("bi");
        assertEquals(2, result.length);
        assertArrayEquals(new String[]{"bi", "ib"}, result);
    }

    @Test
    public void test_3_letters() {
        String[] result = instance.anagrams("bir");
        assertEquals(6, result.length);
        assertArrayEquals(new String[]{"bir", "bri", "ibr", "irb", "rbi", "rib"}, result);
    }

    @Test
    public void test_4_letters() {
        String[] result = instance.anagrams("biro");
        assertEquals(24, result.length);
        assertArrayEquals(new String[]{"biro", "bior", "brio", "broi", "boir", "bori"}, Arrays.copyOfRange(result,0 ,6));
    }
}

File: my_kotlin_dojo/src/anagrams/Kata1Test.kt
package anagrams

import org.junit.After
import org.junit.Before

import org.junit.Assert.*
import org.junit.Test

class Kata1Test {
    var instance : Kata1 = Kata1()
    @Before
    fun setUp() {
    }

    @After
    fun tearDown() {
    }

    @Test
    fun test_2_letters() {
        val result = instance.anagrams("ab")
        assertEquals(2, result.size)
        assertArrayEquals(arrayOf("ab", "ba"), result)
    }

    @Test
    fun test_3_letters() {
        val result = instance.anagrams("abc")
        assertEquals(6, result.size)
        assertArrayEquals(arrayOf("abc", "acb", "bac", "bca", "cab", "cba"), result)
    }

    @Test
    fun test_4_letters() {
        val result = instance.anagrams("abcd")
        assertEquals(24, result.size)
        assertArrayEquals(arrayOf("abcd", "abdc", "acbd", "acdb", "adbc", "adcb"), result.sliceArray(0..5))
    }

    @Test
    fun test_5_letters() {
        val result = instance.anagrams("abcde")
        assertEquals(120, result.size)
    }
}